script-monitoring/constraint_enforcer.py

```
@@ -0,0 +1,345 @@
+        """Get smart recommendation with concrete numbers and actionable advice"""
+        remaining = self.MAX_LINES - core_lines
+        red_threshold = int(self.BASE_LINES * 1.2)
+        to_red = red_threshold - core_lines
+        if level == "green":
+            return f"✅ Continue development ({remaining} lines remaining to max)"
+        elif level == "yellow":
+            return f"⚠️ WARNING: Plan refactoring soon ({remaining} lines remaining, {to_red} until RED)"
+        elif level == "orange":
+            suggestion = f"Increase BASE_LINES to {core_lines + 50}, MAX_LINES to {core_lines + 200}" if to_red < 50 else "Refactor largest files"
+            largest_files = sorted([(k, v['lines']) for k, v in (file_status or {}).items() if isinstance(v.get('lines'), int)], key=lambda x: x[1], reverse=True)[:2]
+            files_info = f" (largest: {', '.join([f'{f}:{l}' for f, l in largest_files])})" if largest_files else ""
```

Ruff (0.12.2)

Ambiguous variable name: `l`

(E741)